frontend/packages/dev-console/src/components/pipelines/__tests__/PipelineDetailsPage.spec.tsx

```diff
@@ -0,0 +1,36 @@
+import * as React from 'react';
+import { shallow } from 'enzyme';
+import { referenceForModel } from '@console/internal/module/k8s';
+import { ErrorPage404 } from '@console/internal/components/error';
+import { PipelineModel } from '../../../models';
+import PipelineDetailsPage from '../PipelineDetailsPage';
+
+const pipelineDetailsProps: React.ComponentProps<typeof PipelineDetailsPage> = {
+  match: {
+    isExact: true,
+    path: `/k8s/ns/:ns/${referenceForModel(PipelineModel)}/:name/`,
+    url: 'k8s/ns/tekton-pipelines/tekton.dev~v1alpha1~Pipeline/simple-pipeline',
+    params: {
+      ns: 'tekton-pipelines',
+    },
+  },
+  kind: 'Pipeline',
+};
+
+const expectedTabs: string[] = [`Details`, `YAML`, `Pipeline Runs`, `Parameters`, `Resources`];
+
+describe('Pipeline details page tests', () => {
+  const pipelineRunDetailsWrapper = shallow(<PipelineDetailsPage {...pipelineDetailsProps} />);
+  it('should render `DetailsPage` Component with proper page tabs', () => {
+    expect(pipelineRunDetailsWrapper.find('DetailsPage').exists()).toBe(true);
+    const pages = pipelineRunDetailsWrapper.find('DetailsPage').prop('pages');
+    expectedTabs.forEach((tab, i) => {
+      expect(pages[i].name).toBe(tab);
+    });
+  });
+
+  it('should render `errorPage` Component ', () => {
+    pipelineRunDetailsWrapper.setState({ errorCode: 404 });
+    expect(pipelineRunDetailsWrapper.find(ErrorPage404).exists()).toBe(true);
```

Comment:
Tiny nit, although I think this is an appropriate test... could you add a second assertion to this test that just says we don't get the Details page. Just in case someone refactors and somehow cases both to render.

Reply:
Yup, a second check would be helpful. Added it.